### DISCUSSION TITLE
Multiline inline tag

### DISCUSSION THREAD

**Discussion by mfix-stripe** [markdoc/markdoc#518] [Q&A] [answered]
yo @segphault @matv-stripe is it possible to add an inline tag that contains multiple lines of text within the tag? 

Take for example I want to add an aside to a piece of text, and bind those together with a single tag, e.g.

```md
I want to annotate {% annotation %}this piece of text{% slot "aside" %}
## My aside has complex content in it

Like a caption! {% .caption %} 
{% /slot %} 
{% /annotation %} test
```

Here is a Sandbox example showing the failure: https://markdoc.dev/sandbox?mode=preview&c=SSB3YW50IHRvIGFkZCBhIG11bHRpbGluZSB7JSBjYWxsb3V0IHR5cGU9ImNoZWNrIiAlfQpMaW5lIDEKCkxpbmUgMgp7JSAvY2FsbG91dCAlfSB0ZXN0Cg%253D%253D

**Comment by segphault** [comment, marked as answer]
No, nesting block content inside of an inline element breaks the document model in a bunch of ways, there's not really a practical way to do that. What I'd recommend for your use case would be declaring the block-level content in its own separate tag and then referencing it in the inline, something like this:

```
{% aside #foo %}
## My aside has complex content in it

Like a caption! {% .caption %} 
{% /aside %}

I want to annotate {% annotation "foo" %}this piece of text{% /annotation %} test
```

In order to use the aside in the annotation tag's render function you would probably have to do a `walk` pass before transform that identifies all of the asides and puts them somewhere in the config, keyed by their IDs. Then you can consume them from the config during transform.

**Reply by mfix-stripe** [reply]
Thanks Ryan 👍
